### PR TITLE
Add auto pop up offerings.

### PR DESCRIPTION
**Pull request by dzeckelev**
Changes:
1) Add `DB.offering.auto_pop_up`.
2) Add `offering.autopopup` setting.
3) Move `EthBackend` from `proc/worker` to `proc/adapter` pkg.
4) Add `proc/looper` pkg for to generate and add jobs on schedule.
5) Add `AutoOfferingPopUp` function to generate `agentPreOfferingPopUp` jobs.

Resolves BV-897.

### CODE REVIEW

File: proc/looper/offering.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+	"github.com/privatix/dappctrl/data"
+	"github.com/privatix/dappctrl/proc/adapter"
+	"github.com/privatix/dappctrl/util/log"
+)
+
+// AutoOfferingPopUp creates AgentPreOfferingPopUp jobs for active offerings.
+// Taken into account a current balance of agent ETH. If ETH is not enough
+// to pop upped all offerings, then not one of its is pop upped. Function
+// calculates the pop up time.
```
Comment (furkhat) [inline proc/looper/offering.go]: `..is not enough to pop up all offerings, then no one of them is popped up..`

File: proc/looper/offering.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	logger.Debug("started AutoOfferingPopUp")
+	do, err := data.ReadBoolSetting(
+		db.Querier, data.SettingOfferingAutoPopUp)
+	if err != nil {
+		logger.Warn(err.Error())
+	}
+
+	var jobs []*data.Job
+
+	if do {
```
Comment (furkhat) [inline proc/looper/offering.go]: To eliminate redundant work and make the use of setting more transparent I suggest to move checking `offering.autopopup` upper.

File: main.go [hunk 3]
```diff
@@ -165,6 +173,39 @@ func createUIServer(conf *ui.Config, logger log.Logger, db *reform.DB,
...
+	if err != nil {
+		return err
+	}
+
+	autoPopUpOfferingFunc := func() []*data.Job {
+		return looper.AutoOfferingPopUp(
+			logger, serviceContractABI, db, ethBack, time.Now)
+	}
+
+	looper.Loop(ctx, logger, db, queue, timeout, autoPopUpOfferingFunc)
```
Comment (furkhat) [inline main.go]: no need to loop if `offering.autopopup` is false.

Comment (dzeckelev) [reply main.go]: What if I want to turn on auto popup later?
There is no need to check the `offering.autopopup` setting.

File: proc/looper/offering.go (hunk 2 above)
Comment (furkhat) [reply proc/looper/offering.go]: Yes, you are right. Didn't think about it.

File: proc/looper/offering.go (hunk 1 above)
Comment (dzeckelev) [reply proc/looper/offering.go]: @furkhat updated.

**Pull request merged**